File: totaleconomy-bukkit/src/main/java/com/ericgrandt/totaleconomy/commonimpl/BukkitPlayer.java
package com.ericgrandt.totaleconomy.commonimpl;

import com.ericgrandt.totaleconomy.common.game.CommonPlayer;
import java.util.UUID;

import net.kyori.adventure.bossbar.BossBar;
import net.kyori.adventure.text.Component;
import org.bukkit.entity.Player;
import org.jetbrains.annotations.NotNull;

public record BukkitPlayer(Player player) implements CommonPlayer {
    @Override
    public void sendMessage(@NotNull Component message) {
        player.sendMessage(message);
    }

    @Override
    public UUID getUniqueId() {
        return player.getUniqueId();
    }

    @Override
    public String getName() {
        return player.getName();
    }

    @Override
    public boolean isNull() {
        return player == null;
    }

    @Override
    public void showBossBar(BossBar bossBar) {
        player.showBossBar(bossBar);
    }

    @Override
    public void hideBossBar(BossBar bossBar) {
        player.hideBossBar(bossBar);
    }
}

File: totaleconomy-paper/src/main/kotlin/com/ericgrandt/totaleconomy/impl/BukkitPlayer.kt
package com.ericgrandt.totaleconomy.impl

import com.ericgrandt.totaleconomy.game.CommonPlayer
import net.kyori.adventure.text.Component
import org.bukkit.entity.Player
import java.util.UUID

class BukkitPlayer : CommonPlayer {
    private var player: Player

    constructor(player: Player) {
       this.player = player
    }

    override fun sendMessage(message: Component) {
        player.sendMessage(message)
    }

    override fun getUniqueID(): UUID {
        return player.uniqueId
    }

    override fun getName(): String {
        return player.name
    }
}